libcore/src/geometry/helper/CorrectGeometry.cpp

```diff
@@ -0,0 +1,354 @@
+#include "geometry/Crossing.h"     // for Crossing
+#include "geometry/Line.h"         // for Line
+#include "geometry/Point.h"        // for Point, Distance
+#include "geometry/Room.h"         // for Room
+#include "geometry/SubRoom.h"      // for SubRoom
+#include "geometry/Transition.h"   // for Transition
+#include "geometry/Wall.h"         // for Wall
+
+#include <algorithm> // for max, find, find_if, sort
+#include <chrono>    // for operator-, duration, high_resolut...
+#include <cmath>     // for isnan
+#include <map>       // for map
+#include <memory>    // for shared_ptr, __shared_ptr_access
+#include <optional>  // for optional, nullopt
+#include <ratio>     // for milli
+#include <stdexcept> // for runtime_error
+#include <string>    // for basic_string, operator+
+#include <utility>   // for pair
+#include <vector>    // for vector<>::iterator, vector<>::con...
+
+namespace geometry::helper
+{
+namespace
+{
+bool RemoveOverlappingWall(const Line & exit, SubRoom & subroom)
+{
+    auto isOverlapping = [&exit](const auto & wall) {
+        return wall.NearlyInLineSegment(exit.GetPoint1()) &&
+               wall.NearlyInLineSegment(exit.GetPoint2());
+    };
+
+    // search for a wall overlapping with the exit
+    const auto & overlappingWallIt =
+        std::find_if(subroom.GetAllWalls().begin(), subroom.GetAllWalls().end(), isOverlapping);
+
+    if(overlappingWallIt == subroom.GetAllWalls().end()) {
+        return false;
+    }
+
+    // Attention: after adding Walls or removing Walls the overlappingWallIt could be invlaid!
+    Wall overlappingWall = *overlappingWallIt;
+
+    double dist_pt1 = (overlappingWall.GetPoint1() - exit.GetPoint1()).NormSquare();
+    double dist_pt2 = (overlappingWall.GetPoint1() - exit.GetPoint2()).NormSquare();
+
+    if(dist_pt1 < dist_pt2) {
+        subroom.AddWall(Wall(overlappingWall.GetPoint1(), exit.GetPoint1()));
+        subroom.AddWall(Wall(overlappingWall.GetPoint2(), exit.GetPoint2()));
+    } else {
+        subroom.AddWall(Wall(overlappingWall.GetPoint1(), exit.GetPoint2()));
+        subroom.AddWall(Wall(overlappingWall.GetPoint2(), exit.GetPoint1()));
+    }
+    subroom.RemoveWall(overlappingWall);
+    return true;
+}
+
+
+/**
+ * @brief Removes doors on walls
+ */
+bool RemoveOverlappingDoors(SubRoom & subroom)
+{
+    Logging::Debug(fmt::format(
+        check_fmt("Enter RemoveOverlappingDoors with SubRoom {}-{}"),
+        subroom.GetRoomID(),
+        subroom.GetSubRoomID()));
+
+    int wallsRemoved = 0; // did we remove anything?
+
+    for(Crossing const * const crossing : subroom.GetAllCrossings()) {
+        wallsRemoved += RemoveOverlappingWall(*crossing, subroom);
+    }
+
+    for(Transition const * const transition : subroom.GetAllTransitions()) {
+        wallsRemoved += RemoveOverlappingWall(*transition, subroom);
+    }
+
+    return wallsRemoved > 0;
+}
+
+
+std::optional<Point> GetSplitPoint(const Wall & wall, const Line & line)
```

Comment:
How about `FindSplitPoint` or `ComputeSplitPoint`?

Reply:
I used `ComputeSplitPoint` to avoid confusion about find returning an iterator...